### PR TITLE
Added failing test case for when column names are added

### PR DESCRIPTION
**Pull request by asgrim**
Adds a failing test for bug report #11

### CODE REVIEW

**Comment by asgrim** [comment]
Not entirely sure how to fix this, I tried changing the pattern:

```diff
diff --git a/main/dumpscanner.l b/main/dumpscanner.l
index b40d21c..b3b4473 100644
--- a/main/dumpscanner.l
+++ b/main/dumpscanner.l
@@ -94,7 +94,7 @@ bool is_anonymized() {
                                       return CREATE_TABLE;
                                     }
                                   }
-^(INSERT|REPLACE)\ INTO\ `[^`]+`  { DUPOUT
+^(INSERT|REPLACE)\ INTO\ `[^`]+`(?:\ \([^\)]+\))?   { DUPOUT
                                     if (is_anonymized()) {
                                       BEGIN(ST_VALUES);
                                       DEBUG_MSG("\nENTERING STATE ST_VALUES")
```

But this will not satisfy the cases where the order of columns is different, or where less columns are specified than exist

**Comment by ppomes** [comment]
Currently : 
- column order is computed when parsing the CREATE TABLE statement (ie: of you need to anonymize the 'a' column, and the 'a' column is in second position, the second field on each insert statement will be anonymized). 
- As far as I know, mysqldump cannot extract only some columns, so the number of fields (from 'insert' statements) should match the number of columns (from 'create table' statement). And I also do not know a way of changing fields order.

So when using Myanon only with result of mysqldump comands, your fix should work - I suppose you edited manually the test file?

**Comment by asgrim** [comment]
> So when using Myanon only with result of mysqldump comands, your fix should work - I suppose you edited manually the test file?

Yeah - the suggested fix works as long as the number of columns matches (as per a direct `mysqldump` without any changes). And yes, I thought of the additional cases and added them manually, since they're valid in general. If you're happy for me to comment them out with an explanation that those aren't supported cases at the moment, I'll update this patch :smile:

**Comment by ppomes** [comment]
Yes, that would be nice!

File: tests/test1.conf [hunk 1]
```diff
@@ -26,4 +26,8 @@ tables = {
    `the_blobs` = {
```
Comment (asgrim) [inline tests/test1.conf:26]: Hmm weird, the `the_blobs` configuration/table seems to be breaking this fix, it passes if I remove this part:

```diff
diff --git a/tests/test1.conf b/tests/test1.conf
index 5827ba6..07f1bc3 100644
--- a/tests/test1.conf
+++ b/tests/test1.conf
@@ -23,9 +23,6 @@ tables = {
      `c`    = fixed ''
      `name` = texthash 10
    }
-   `the_blobs` = {
-     `blob1` = fixed unquoted '0x676f6f64627965'
-   }
    `test_with_column_names` = {
      `a` = fixed '111'
      `b` = fixed '222'
diff --git a/tests/test1_anon.sql b/tests/test1_anon.sql
index ee15549..1c1c47b 100644
--- a/tests/test1_anon.sql
+++ b/tests/test1_anon.sql
@@ -143,7 +143,7 @@ CREATE TABLE `the_blobs` (
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
 
 INSERT INTO `the_blobs` VALUES (
-  0x676f6f64627965,
+  0x68656c6c6f,
   0x68656c6c6f,
   0x68656c6c6f,
   0x68656c6c6f
```

Not entirely sure why at the moment :thinking:

Comment (ppomes) [reply tests/test1.conf:26]: I will have a look later today, stay tuned;-)

Comment (asgrim) [reply tests/test1.conf:26]: I appreciate it, thank you!

Comment (ppomes) [reply tests/test1.conf:26]: I was able to run your patch successfully, using the following test1.zip (test1.sql/test1_anon.sql files). The test1.sql is a little bit closer to the output of a real mysqldump command. 

I have plan to review flex state transitions to have better support for those manual editions (but I think this is out of scope of this PR).

In the meanwhile, I think you can just update the PR with theses two files, and the 'make check' will be ok ;-)

[test1.zip](https://github.com/ppomes/myanon/files/9143460/test1.zip)

Comment (ppomes) [reply tests/test1.conf:26]: I just updated the tests files, CI is now ok!

Comment (asgrim) [reply tests/test1.conf:26]: Thanks - sorry I didn't get to it yet! :+1:

**Pull request merged**